torch_xla/experimental/tpu.py

```diff
@@ -70,17 +71,24 @@ def _get_metadata(key: str) -> str:
   return resp.text
 
 
-def process_bounds_size(default: int = 1) -> int:
-  """Returns number of processes across all TPU hosts."""
+def process_bounds_size() -> Optional[int]:
+  """Returns number of processes across all TPU hosts, or None if unknown."""
   process_bounds = xu.getenv_as(xenv.TPU_PROCESS_BOUNDS, str)
-  return MeshShape.from_string(
-      process_bounds).size if process_bounds else default
+  return MeshShape.from_string(process_bounds).size if process_bounds else None
 
 
-def num_local_processes(local_chips: int = 4) -> int:
+def num_available_chips() -> int:
+  """Returns the number of local chips in /dev/"""
+  return len(glob.glob('/dev/accel?'))
```

Comment:
what's gonna happen if it is not on a TPU device?

Reply:
`tpu.num_available_chips` will return `0` in that case

Reply:
Seems a bit hacky? Is there an API to allow you do that? The `/dev/accel?` could change or could be non-TPU.

Reply:
You can query PJRT for the number of devices, but we actually have to run this check before we initialize PJRT, since it tells us how many PJRT clients to create.

I agree this is a little hacky, but it helps us handle the case where you pass through one TPU device to a container out of multiple, compared to creating a mapping between TPU types and local chips. FWIW, `/dev/accel*` is hardcoded  within `libtpu` and the TPU driver as well, so I don't think that convention is likely to change without us knowing.